app/src/main/java/org/ergoplatform/android/wallet/WalletDbDao.kt

```diff
@@ -12,21 +12,40 @@ interface WalletDbDao {
     @Query("DELETE FROM wallet_configs WHERE id = :walletId")
     suspend fun deleteWalletConfig(walletId: Int)
 
-    @Query("DELETE FROM wallet_states WHERE public_address = :publicAddress")
-    suspend fun deleteWalletState(publicAddress: String)
+    @Query("DELETE FROM wallet_states WHERE wallet_first_address = :firstAddress")
+    suspend fun deleteWalletState(firstAddress: String)
+
+    @Query("DELETE FROM wallet_tokens WHERE wallet_first_address = :publicAddress")
+    suspend fun deleteTokensByAddress(publicAddress: String)
```

Comment:
shouldn't the parameter be renamed firstAddress?

Reply:
Good catch, thank you! Even worse, it was selecting for the wrong field! Did not notice because there are no derived addresses in use yet.